pkg/sbom/cyclonedx/unmarshal.go

```diff
@@ -341,6 +341,11 @@ func toPackage(component cdx.Component) (bool, string, *ftypes.Package, error) {
 	}
 
 	pkg := p.Package()
+	// Trivy's marshall loses case-sensitivity in PURL used in SBOM for Go,
+	// so we have to use an original package name
+	if p.PackageType() == ftypes.GoBinary {
```

Comment:
I'm wondering if this problem is applied to other languages as well. What if removing the if-statement?

Reply:
you're right. there are uses `ToLower` for `PyPI` and `npm`, so it can arise such issue for them.
let's try to remove this `if` statement